services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/oauth/OpenIDConnectManagerImpl.java

```diff
@@ -126,13 +128,21 @@ public void setClaimProviders(Map<OIDCClaimName, OIDCClaimProvider> claimProvide
 	}
 
 	public static void validateAuthenticationRequest(OIDCAuthorizationRequest authorizationRequest, OAuthClient client) {
-		ValidateArgument.validUrl(authorizationRequest.getRedirectUri(), "Redirect URI");
+		try {
+			ValidateArgument.validUrl(authorizationRequest.getRedirectUri(), "Redirect URI");
+		} catch (IllegalArgumentException e) {
+			throw new OAuthBadRequestException(OAuthErrorCode.invalid_request, e.getMessage());
```

Comment:
throw new OAuthBadRequestException(OAuthErrorCode.invalid_request, e), i.e., chain the exceptions